treeTrimming.js

```diff
@@ -99,3 +128,63 @@ function trimRedundantNodes(inputTree){
 	}
 	return trimInner(tree);
 }
+
+function trimAttributedNodes(inputTree, attribute){
```

Comment:
I think trimNodesWithAttribute() would be a better function name.

Reply:
This code was not written by Ed, possibly was written by Max. Not used by markMinMax(). Unclear whether it works or is used anywhere.